core/src/validator.rs

```diff
@@ -2383,8 +2401,16 @@ fn wait_for_supermajority(
                 sleep(Duration::new(1, 0));
             }
             rpc_override_health_check.store(false, Ordering::Relaxed);
+            if config.wait_for_supermajority.is_none() {
+                if let Some(proto_path) = &config.wen_restart_proto_path {
+                    if let Err(e) = wen_restart_mark_done(proto_path) {
```

Comment:
Hmm, if we need two back to back wen_restart in the future, I'd be really worried.

This is similar to --wait_for_supermajority, you have to specify --wen_restart on command line with your desired proto path. So we can suggest that you put date in the proto path, so you never have to delete the file. After wen_restart finishes correctly you should remove --wen_restart from your command line, as you do in --wait_for_supermajority. You can keep it there forever, the proto path will have state DONE so it will do nothing. During next --wen_restart we just use a new path with different dates so we don't mix data from two wen_restarts together.